xds/internal/balancer/clusterresolver/priority_test.go

```diff
@@ -1,834 +0,0 @@
-	xdsC := fakeclient.NewClientWithName(testBalancerNameFooBar)
-	cc := testutils.NewTestClientConn(t)
-	builder := balancer.Get(Name)
-	edsb := builder.Build(cc, balancer.BuildOptions{Target: resolver.Target{URL: *testutils.MustParseURL("dns:///" + testEDSService)}})
-	if edsb == nil {
-		t.Fatalf("builder.Build(%s) failed and returned nil", Name)
-	}
-	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
-	defer cancel()
-	if err := edsb.UpdateClientConnState(balancer.ClientConnState{
-		ResolverState: xdsclient.SetClient(resolver.State{}, xdsC),
-		BalancerConfig: &LBConfig{
-			DiscoveryMechanisms: []DiscoveryMechanism{{
-				Cluster: testClusterName,
-				Type:    DiscoveryMechanismTypeEDS,
-			}},
-			xdsLBPolicy: *wrrLocalityLBConfig,
-		},
-	}); err != nil {
-		edsb.Close()
-		t.Fatal(err)
-	}
-	if _, err := xdsC.WaitForWatchEDS(ctx); err != nil {
-		edsb.Close()
-		t.Fatalf("xdsClient.WatchEndpoints failed with error: %v", err)
-	}
-	return edsb, cc, xdsC, func() {
-		edsb.Close()
-	}
-}
-
-// When a high priority is ready, adding/removing lower locality doesn't cause
-// changes.
-//
-// Init 0 and 1; 0 is up, use 0; add 2, use 0; remove 2, use 0.
-func (s) TestEDSPriority_HighPriorityReady(t *testing.T) {
-	edsb, cc, xdsC, cleanup := setupTestEDS(t, nil)
-	defer cleanup()
-
-	// Two localities, with priorities [0, 1], each with one backend.
-	clab1 := xdstestutils.NewClusterLoadAssignmentBuilder(testClusterNames[0], nil)
-	clab1.AddLocality(testSubZones[0], 1, 0, testEndpointAddrs[:1], nil)
-	clab1.AddLocality(testSubZones[1], 1, 1, testEndpointAddrs[1:2], nil)
-	xdsC.InvokeWatchEDSCallback("", parseEDSRespProtoForTesting(clab1.Build()), nil)
-
-	addrs1 := <-cc.NewSubConnAddrsCh
-	if got, want := addrs1[0].Addr, testEndpointAddrs[0]; got != want {
-		t.Fatalf("sc is created with addr %v, want %v", got, want)
-	}
-	sc1 := <-cc.NewSubConnCh
-
-	// p0 is ready.
-	edsb.UpdateSubConnState(sc1, balancer.SubConnState{ConnectivityState: connectivity.Connecting})
-	edsb.UpdateSubConnState(sc1, balancer.SubConnState{ConnectivityState: connectivity.Ready})
-
-	// Test roundrobin with only p0 subconns.
-	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
-	defer cancel()
-	if err := cc.WaitForRoundRobinPicker(ctx, sc1); err != nil {
-		t.Fatal(err)
-	}
-
-	// Add p2, it shouldn't cause any updates.
-	clab2 := xdstestutils.NewClusterLoadAssignmentBuilder(testClusterNames[0], nil)
-	clab2.AddLocality(testSubZones[0], 1, 0, testEndpointAddrs[:1], nil)
-	clab2.AddLocality(testSubZones[1], 1, 1, testEndpointAddrs[1:2], nil)
-	clab2.AddLocality(testSubZones[2], 1, 2, testEndpointAddrs[2:3], nil)
-	xdsC.InvokeWatchEDSCallback("", parseEDSRespProtoForTesting(clab2.Build()), nil)
-
-	select {
-	case <-cc.NewSubConnCh:
-		t.Fatalf("got unexpected new SubConn")
-	case <-cc.RemoveSubConnCh:
-		t.Fatalf("got unexpected remove SubConn")
-	case <-time.After(defaultTestShortTimeout):
-	}
-
-	select {
```

Comment:
This line and below isn't captured in priority_test TestPriority_HighPriorityReady. Can you please copy paste this over there?

Reply:
Added a couple of lines in there to check that p0 subConns are still in use, as long as p0 is ready.